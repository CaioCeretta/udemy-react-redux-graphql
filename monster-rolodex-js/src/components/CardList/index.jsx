import { CardItem } from "../CardItem";
import "./styles.css";

export const CardList = ({monsters}) => {
    
    return (
      <div className="card-list">
        {
          monsters.map((monster) => {
            const {id, name, email} = monster

          return (
          
            <CardItem key={id} id={id} name={name} email={email} />
            /* <img src={`https://www.robohash.org/${monster.id}?set=2&size=180x180`} alt={`monster ${name}`} />
            <h1>{name}</h1>
            <p>{email}</p> */
          
        )
        })}
      </div>
    );
}
