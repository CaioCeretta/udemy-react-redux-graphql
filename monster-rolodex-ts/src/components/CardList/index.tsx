import { Monster } from "../../App";
import { CardItem } from "../CardItem";
import "./styles.css";

type CardProps = {
  monsters: Monster[];
}


export const CardList = ({ monsters }: CardProps) => {

  return (
    <div className="card-list">
      {
        monsters.map((monster) => {
          const { id, name, email } = monster

          return (

            <CardItem key={id} monster={monster} />
            /* <img src={`https://www.robohash.org/${monster.id}?set=2&size=180x180`} alt={`monster ${name}`} />
            <h1>{name}</h1>
            <p>{email}</p> */

          )
        })}
    </div>
  );
}
