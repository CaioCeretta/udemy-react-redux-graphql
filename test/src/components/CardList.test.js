import { shallow } from "enzyme";

import CardList from './CardList'

it('expect to render the CardList component', () => {
  const mockRobots = [
    {
      id: 1,
      name: 'Jorge',
      username: 'JorJor',
      email: 'jorge@gmail.com'
    }
  ]
  expect(shallow(<CardList robots={mockRobots}/>)).toMatchSnapshot()

  

})