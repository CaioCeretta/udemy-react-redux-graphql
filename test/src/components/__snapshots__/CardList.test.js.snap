// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render the CardList component 1`] = `
<div>
  <Card
    email="jorge@gmail.com"
    id={1}
    key="0"
    name="Jorge"
  />
</div>
`;
